utilities/src/serialize/impls.rs

```diff
@@ -398,6 +398,45 @@ impl<T: CanonicalDeserialize> CanonicalDeserialize for Vec<T> {
     }
 }
 
+impl<T: CanonicalDeserialize> CanonicalDeserialize for [T; 32] {
+    #[inline]
+    fn deserialize_with_mode<R: Read>(
+        mut reader: R,
+        compress: Compress,
+        validate: Validate,
+    ) -> Result<Self, SerializationError> {
+        let len = u64::deserialize_with_mode(&mut reader, compress, validate)?;
+        assert!(len == 32); // in the future we can parametrize the array on length
+        let mut values = vec![];
+        for _ in 0..len {
+            values.push(T::deserialize_with_mode(&mut reader, compress, Validate::No)?);
+        }
+
+        if let Validate::Yes = validate {
+            T::batch_check(values.iter())?
+        }
+        let arr = values.try_into()
+            .unwrap_or_else(|values: Vec<T>| panic!("Expected a Vec of length {} but it was {}", 32, values.len()));
+
+        Ok(arr)
```

Comment:
We don't need to use `vec![]` here; we can use https://doc.rust-lang.org/std/primitive.array.html#method.map

Reply:
So i.e. something like `[(); 32].map(our_fn_here)`

Reply:
Done. Note that I `collect` into a Vec first so the `Result`s automatically bubbles up.
```
        let values = [(); 32].iter().map(|_| {
            T::deserialize_with_mode(&mut reader, compress, Validate::No)
        }).collect::<Result<Vec<T>,_>>()?;
```